src/plugins/controls/public/control_group/external_api/control_group_renderer.tsx

```diff
@@ -87,7 +89,8 @@ export const ControlGroupRenderer = forwardRef<AwaitingControlGroupAPI, ControlG
             ...initialInput,
           },
           undefined,
-          settings
+          settings,
```

Comment:
nit: Is there any reason why the `fieldFilterPredicate` is outside of the `settings` instead of being wrapped inside it like it used to be? It might require fewer changes if it was left inside the settings object.

Reply:
@ThomThomson , actually, it is because the whole settings is being passed to redux, which created the problem in the first place. So instead of manipulating the settings object, I kept it saperate. Let me know your thoughts.

https://github.com/elastic/kibana/blob/94c9491748da312b2bbb0e8716f92e62db7659b5/src/plugins/controls/public/control_group/embeddable/control_group_container.tsx#L120-L126

Reply:
Ah, that makes perfect sense. Thanks for the explanation!